cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -1855,19 +1854,25 @@ class ArraySelecter : public TypeVisitor {
         ctx_(ctx),
         array_(array),
         k_(options.k),
+        order_(options.sort_keys[0].order),
```

Comment:
This assumes that there is at least one key. If `sort_keys` is empty, it would segfault. Unless the options are validated somewhere else.
I recommend to check this explicitly in the constructor.

Reply:
Similarly, for the other `XXXSelecters`.

Reply:
the other XXXSelecters doesn't need these because for RecordBatch and Table it's asume that there is more than two columns so there is not a global order for all of them, each column has their own order determined in options.sort_keys [(key_name, sort_order)].